pylint/checkers/classes.py

```diff
@@ -995,16 +995,22 @@ class SpecialMethodsChecker(BaseChecker):
                   'invalid number of parameters. If it has too few or '
                   'too many, it might not work at all.',
                   {'old_names': [('E0235', 'bad-context-manager')]}),
+        'E0303': ('__len__ does not return non-negative integer',
+                  'bad-len-returned',
```

Comment:
They will probably have different names, so that they can be easily disabled on a case by case basis. How about invalid-length-returned, does that sound better?